apps/helpers.py

```diff
@@ -556,3 +561,109 @@ def validate_docker_image(image: str):
         )
+
+
+def generate_schema_org_description(app_instance: BaseAppInstance) -> str:
+    """Generate schema.org structured data for App, User, and Project models."""
+
+    # Safely get related objects
+    try:
+        user_instance = User.objects.get(id=app_instance.owner_id)
+    except User.DoesNotExist:
+        raise ValueError(f"User with id {app_instance.owner_id} does not exist")
+
+    try:
+        project_instance = Project.objects.get(id=app_instance.project_id)
+    except Project.DoesNotExist:
+        raise ValueError(f"Project with id {app_instance.project_id} does not exist")
+
+    # Convert models to dictionaries with safe defaults
+    app_data = model_to_dict(app_instance, exclude=["_state"])
+    user_data = model_to_dict(user_instance, exclude=["_state", "password"])
+    project_data = model_to_dict(project_instance, exclude=["_state"])
+
+    # Safely add special fields
+    app_data.update(
+        {"k8s_values": app_instance.k8s_values or {}, "info": app_instance.info or {}, "url": app_instance.url or {}}
+    )
+
+    # Build software requirements safely
+    software_requirements = {}
+    if app_data["k8s_values"]:
+        requests = requests = app_data["k8s_values"].get("flavor", {}).get("requests", {})
+        limits = app_data["k8s_values"].get("flavor", {}).get("limits", {})
+
+        # Resource mapping: (schema_field_name, k8s_field_name)
+        resource_mapping = [
+            ("cpu", "cpu"),
+            ("gpu", "nvidia.com/gpu"),
+            ("memory", "memory"),
+            ("storage", "ephemeral-storage"),
+        ]
+
+        for field_name, k8s_name in resource_mapping:
+            if requests.get(k8s_name):
+                software_requirements[f"{field_name}Request"] = requests[k8s_name]
+            if limits.get(k8s_name):
+                software_requirements[f"{field_name}Limit"] = limits[k8s_name]
+
+    schema = {
+        "@context": "https://schema.org",
+        "@type": "Dataset",
+        "name": "Application Deployment Metadata",
+        "description": "Structured metadata for deployed applications, users, and projects",
+        "dateCreated": timezone.now().isoformat(),
+        "creator": {"@type": "Organization", "name": "SciLifeLab Data Centre", "url": "https://www.scilifelab.se"},
+        "hasPart": [
+            {
+                "@type": "SoftwareApplication",
+                "name": app_data.get("name"),
+                "url": app_data.get("url") if app_data.get("url") else None,
+                "softwareVersion": app_data.get("chart"),
+                "sourceCodeURL": app_data.get("source_code_url"),
+                "dateCreated": app_data.get("created_on").isoformat() if app_data.get("created_on") else None,
+                "dateModified": app_data.get("updated_on").isoformat() if app_data.get("updated_on") else None,
+                "applicationCategory": "Cloud Application",
+                "operatingSystem": "Kubernetes",
```

Comment:
I have nothing against it, just curious, did you see anyone else setting these keys to the specified values?:)